addons/mail/models/mail_render_mixin.py

```diff
@@ -332,6 +337,40 @@ def _render_template_qweb(self, template_src, model, res_ids,
 
         return results
 
+    @api.model
+    def _render_template_qweb_static(self, template_src, model, res_ids):
+        """Render the template with regex instead of qweb to avoid `eval` call."""
```

Comment:
Maybe worth a comment telling 'supporting only qweb allowed expression, no custom variables in that mode'